sklearn/utils/estimator_checks.py

```diff
@@ -3013,6 +3013,8 @@ def check_no_attributes_set_in_init(name, estimator_orig):
 
     # Test for no setting apart from parameters during init
     invalid_attr = set(vars(estimator)) - set(init_params) - set(parents_init_params)
+    # Ignore private attributes
+    invalid_attr = set([attr for attr in invalid_attr if not attr.startswith("_")])
```

Comment:
Why is this needed?

Reply:
In the test below you refer to private class attribs, but here you use `vars` which will only get instance attribs?

Reply:
Added clarification to the test:

```
    # also check if cloning an estimator which has non-default set requests is
    # fine. Setting a non-default value via `set_{method}_request` sets the
    # private _metadata_request instance attribute which is copied in `clone`.
    check_no_attributes_set_in_init(
        "estimator_name", ConformantEstimatorClassAttribute().set_fit_request(foo=True)
    )
```

removing this line would make this test fail. The question is if we want to skip all private attributes or only `_metadata_request`.